### PR TITLE
File format

### PR DESCRIPTION
**Pull request by joachimwolff**
* [x] Flake8 passes (`flake8 . --exclude=.venv,.build,planemo_test_env,build --ignore=E501,F403,E402,F999,F405,E712`)
* [x] Local tests pass (`py.test hicexplorer --doctest-modules`)

This PR reorganizes how file formats are handled to an object oriented structure. It makes it easier to extend and the code better readable. 

- All handling is in the folder `lib`, the class hiCMatrix is using only the `MatrixFileHandler` to read/write matrices and does not have any source code handling read or write in it anymore. 
- Supported file formats for all tools are `h5` and `cool`. The support for all other file formats is dropped.
- `hicExport` is replaced by `hicConvertFileFormats`. 
- `hicConvertFileFormats` supports the transfer of `homer`, `h5` and `hicpro` to `cool` and with help of the library `hic2cool` an export of the `hic` format to `cool`. 
- Unused source code was deleted. I hope I did not delete any non-dead src, the test cases pass all, but maybe we have to add deleted src in the future again. 
- Static functions within the class hiCMatrix were reorganized as regular functions of this class. 
- A small fix in hicPlotTads to check if `tracks` in line 1102 is not None. 
- Integration of the test cases from my HiWi, they do not pass so far in the develop branch but they do this here.

### CODE REVIEW

**Comment by vivekbhr** [comment]
Did the export to GInteraction format stopped working, or did you decide to exclude this on purpose?

**Comment by joachimwolff** [comment]
GInteractions is still there but out commented. There is a write function but no read? Maybe I have overseen it but I decided to out comment it at the moment and do the merge. I do this because this branch and the viewpoint branch come both with big number of changes and I want to have both in develop. The reorganized code can be used as a base from now on and this will make future branch merges easier.
Don't worry, GInteractions and mcool will be both supported in the next release.

**Comment by vivekbhr** [comment]
@joachimwolff GInteraction input may or may-not be expanded to full matrix that's why I didn't make it an input format back then, but it's a very useful format nevertheless. Also, apart from GI and mcool, we also need the "raw marix" format as output option since a human-readable text file could always be useful.

File: .travis.yml [hunk 1]
```diff
@@ -54,39 +54,39 @@ jobs:
...
       os: linux
       env: TRAVIS_PYTHON_VERSION=3.6
-      script: py.test hicexplorer/test/long_run/test_buildMatrix.py
+      script: py.test hicexplorer/test/long_run/test_hicBuildMatrix.py
     - stage: test
       os: osx
       env: TRAVIS_PYTHON_VERSION=3.6
-      script: py.test hicexplorer/test/long_run/test_buildMatrix.py
+      script: py.test hicexplorer/test/long_run/test_hicBuildMatrix.py
     - stage: test
```
Comment (fidelram) [inline .travis.yml]: unrelated to the actual PR, but what is the advantage of running each tests separately?

Comment (joachimwolff) [reply .travis.yml]: With the separate runs the test cases are executed (more or less) in parallel and we fulfill the time limits of travis per job. If we don't do this the test cases would be quitted by travis after 40 - 50 minutes.

Comment (fidelram) [reply .travis.yml]: ahhh!! cool

**Pull request merged**